app/models/search_form.rb

```diff
@@ -0,0 +1,35 @@
+class SearchForm
+
+  ADMIN_SEX = %w[A F M N O U]
+
+  attr_reader(*QueryParameters.all)
+
+  # Attribute name must match the key name expected by V2MessageBuilder as
+  # the hash is generated in to_query using attribute names
+  def initialize(params)
+    @patient_name = hash_or_nil(params.slice(:family_name, :given_name))
+    @patient_dob = params[:patient_dob]
+    phone = Phonelib.parse(params[:phone])
+    @phone = hash_or_nil(area_code: phone.area_code, local_number: phone.local_number)
+    @admin_sex = params[:sex] if ADMIN_SEX.include?(params[:sex])
+    @address = hash_or_nil(params.slice(:street, :city, :state, :zip))
+    @id = params[:id]
+    #@mothers_maiden_n = params[:mothers_maiden_name]
+    @mothers_maiden_n = hash_or_nil(params.slice(:mothers_maiden_name, name_type_code: "M"))
```

Comment:
Check to make sure the name and keys of the `mothers_maiden_name` field line up with what is expected from the QBP Client's `query()` function.

You can see an example of what the hash should look like here:
https://github.com/dvci/iis-shc-issuer/blob/main/test/iis_shc_issuer/profile_errors_test.rb#L9

The `given_name` can default to an empty string if it is not provided by a form parameter.

Reply:
Fixed in commit https://github.com/dvci/iis-shc-issuer/pull/10/commits/0a307a0cb19c08543abf7320a6f005bbc44ebec4.